agent/core/src/main/java/org/jolokia/restrictor/RestrictorFactory.java

```diff
@@ -115,9 +142,35 @@ public static PolicyRestrictor lookupPolicyRestrictor(String pLocation) throws I
                 is = RestrictorFactory.class.getResourceAsStream(path);
             }
         } else {
-            URL url = new URL(pLocation);
-            is = url.openStream();
+            URLConnectionResult urlConnectionResult = buildUrlConnection(pLocation);
```

Comment:
Yeah, it resulted after I figured the Sun implementation for the URLConnection will not expose any "security headers" even if they are set. Extra method then as you describe.